### PR TITLE
fix: avoid NPE in RegenAuthoritySystem

### PR DESCRIPTION
**Pull request by skaldarnar**

### CODE REVIEW

**Comment by skaldarnar** [comment]
The tests were failing before, and this incident is not covered by any test.

File: src/main/java/org/terasology/logic/health/RegenAuthoritySystem.java [hunk 1]
```diff
@@ -122,6 +122,10 @@ private void regenerate(long currentTime) {
         for (EntityRef entity : regenSortedByTime.values()) {
             RegenComponent regen = entity.getComponent(RegenComponent.class);
             HealthComponent health = entity.getComponent(HealthComponent.class);
+            if (regen == null || health == null) {
+                logger.debug("Entity '{}' without RegenComponent or HealthComponent scheduled for regeneration - skipping", entity);
+                continue;
+            }
             if (health != null && health.nextRegenTick < currentTime) {
```
Comment (pollend) [inline src/main/java/org/terasology/logic/health/RegenAuthoritySystem.java]: you don't need this null check after this.

**Review by pollend — CHANGES_REQUESTED** [review]

File: src/main/java/org/terasology/logic/health/RegenAuthoritySystem.java (hunk 1 above)
Comment (skaldarnar) [reply src/main/java/org/terasology/logic/health/RegenAuthoritySystem.java]:
```suggestion
            if (health.nextRegenTick < currentTime) {
```

**Review by pollend — APPROVED** [review]

**Comment by skaldarnar** [comment]
Test failed before.

**Pull request merged**